### ISSUE TITLE
task.unique and task.sleep interaction bug

### ISSUE THREAD

**Issue by dlashua** [custom-components/pyscript#2]
````
@state_trigger("input_boolean.test_office_motion == 'on'")
def office_occupied():
    task.unique('office_occupied')
    log.info('office_occupied running')
    state.set('binary_sensor.test_office_occupied', 'on', {'held': 'yes'})
    task.wait_until(state_trigger="input_boolean.test_office_motion == 'off'")
    state.set('binary_sensor.test_office_occupied', 'on', {'held': 'no'})
    task.sleep(5)
    state.set('binary_sensor.test_office_occupied', 'off')
````

To see this in action, turn the input_boolean on. See the binary_sensor turn on. Turn in input_boolean off, then back on again. Wait 5 seconds. The binary_sensor will turn off, but task.unique should have prevented that.

It seems like the sleeping task is not killed by task.unique, so it still runs to completion.

**Comment by craigbarratt** [comment]
I went ahead and released 0.12.  Please re-open this issue if you see the same problem in 0.12.

**Comment by dlashua** [comment]
This seems to be fixed in "master".
